mathics/core/symbols.py

```diff
@@ -468,30 +382,57 @@ def format(self, evaluation, form, **kwargs) -> "BaseExpression":
         )
         return result
 
-    def is_free(self, form, evaluation) -> bool:
+    def get_atoms(self, include_heads=True):
         """
-        Check if self has a subexpression of the form `form`.
+        Returns a list of atoms that appears in the expression.
         """
-        from mathics.builtin.patterns import item_is_free
+        # Comment @mmatera:
+        # This function is used just in Graphics.apply_makeboxes
+        # to check if a graphics expression is composed just by
+        # real numbers (or integer) or graphics symbols.
+        # Probably, there is a simpler way to implement it without using
+        # this method.
+        return []
 
-        return item_is_free(self, form, evaluation)
+    def get_attributes(self, definitions):
```

Comment:
Here and in other places I think that `@attribute` should be used.

Reply:
I sort of agree but this is a pervasive change and not something to be put into this PR now. 

I had also considered this for `get_head` and `get_elements`. 

I have opened https://github.com/Mathics3/mathics-core/issues/174 to track this. 
Feel free to add to that or correct what is there.